src/GafferImage/OpenImageIOReader.cpp

```diff
@@ -517,6 +533,25 @@ void OpenImageIOReader::hashFileName( const Gaffer::Context *context, IECore::Mu
 	}
 }
 
+std::string OpenImageIOReader::computeDataType( const Gaffer::Context *context ) const
+{
+
+	std::string fileName = fileNamePlug()->getValue();
+	const ImageSpec *spec = imageSpec( fileName, (MissingFrameMode)missingFrameModePlug()->getValue(), this, context );
+	if( !spec )
+	{
+	
+		return "";
+	
+	}
+	const int bitsPerSample = spec->get_int_attribute( "oiio:BitsPerSample", 0 );
+	if( bitsPerSample )
+	{
+		return boost::str( boost::format( "uint%d"  ) % bitsPerSample );
```

Comment:
https://github.com/OpenImageIO/oiio/blob/5b25ba0d36e6031831b7778b6c36423e68f7e8f5/src/tiff.imageio/tiffoutput.cpp#L97

seems to be ok?